src/Postmark/Transport.php

```diff
@@ -294,6 +305,29 @@ private function processHeaders(&$payload, $message) {
 				}
 			}
 		}
+
+		// we process the default headers after, because in an e-mail every
+		// header can be present multiple times $headers is a list and not
+		// a key-value map. The default headers are only added if there is no
+		// header present with the same name one **or** multiple times.
+		//
+		// Default headers do not support being appended to existing headers
+		// with the same name.
+		foreach ($this->defaultHeaders as $header => $value) {
+			if (isset($headersSetInMessage[$header])) {
+				continue;
+			}
+
+			if ($header === 'X-PM-Tag') {
```

Comment:
This logic for processing an individual header can also be found above, on line 282. I'm thinking it might be good for this to be extracted to a separate method, to avoid updating two places in case we'll need additional logic in the future.